crates/hir/src/doc_links.rs

```diff
@@ -0,0 +1,226 @@
+//! Resolves links in markdown documentation.
+
+use std::iter::once;
+
+use itertools::Itertools;
+use url::Url;
+
+use crate::{db::HirDatabase, Adt, AsName, Crate, Hygiene, ItemInNs, ModPath, ModuleDef};
+use hir_def::{db::DefDatabase, resolver::Resolver};
+use syntax::ast::Path;
+
+pub fn resolve_doc_link<T: Resolvable + Clone, D: DefDatabase + HirDatabase>(
```

Comment:
Can we make this a method of `Docs` trait?

Reply:
The docs trait is implemented for `T: Into<AttrDef>` which is a different set of types from those that implement `Resolvable`.

Conceptually they're not really the same thing. `Resolvable` really just means that we can find a base which we can then use to resolve documentation links. There isn't a strict requirement that something implementing `Resolvable` has, or could have, documentation.